bolt-sidecar/src/driver.rs

```diff
@@ -60,15 +63,23 @@ impl fmt::Debug for SidecarDriver<StateClient, BlsSigner, PrivateKeySigner> {
     }
 }
 
-impl SidecarDriver<StateClient, BlsSigner, PrivateKeySigner> {
+impl SidecarDriver<StateClient, PrivateKeySigner> {
```

Comment:
to keep the BLS generic trait in the SidecarDriver, we will need to have 2 separate associated methods `new()`.
One for the private key signer and one for commit-boost. Something like this:

```rust
impl SidecarDriver<StateClient, LocalBlsSigner, PrivateKeySigner> {
	pub fn new(cfg: Config) -> eyre::Result<Self> {
		// ...
		Self::from_components(...)
	}
}

impl SidecarDriver<StateClient, CommitBoostSigner, PrivateKeySigner> {
	pub fn new(cfg:Config) -> eyre::Result<Self> {
		// ...
		Self::from_components(...)
	}
}
```

Reply:
thanks good idea! done